crates/backend/src/encode.rs

```diff
@@ -34,7 +58,45 @@ impl Interner {
     }
 
     fn intern_str(&self, s: &str) -> &str {
-        self.intern(&Ident::new(s, Span::call_site()))
+        let mut strings = self.strings.borrow_mut();
+        if let Some(s) = strings.get(s) {
+            return unsafe { &*(&**s as *const str) };
```

Comment:
Good call. 

This still needs to be somewhat tricky because the main purpose here was to get `&'intern str` (moving the lifetime to the same as `Interner`) and I got a bit carried away with deduplication as well.

Thankfully though you've written a crate that exactly solves this use case: bumpalo!

Reply:
Ha! I didn't think of that :-p